storage_server/report_server.py

```diff
@@ -526,13 +526,19 @@ def addSuppressBug(self, run_id, bugs_to_suppress):
         try:
             suppressList = []
             for bug_to_suppress in bugs_to_suppress:
-                suppress_bug = SuppressBug(run_id,
-                                           bug_to_suppress.bug_hash,
-                                           bug_to_suppress.file_name,
-                                           bug_to_suppress.comment)
-                suppressList.append(suppress_bug)
+                res = self.session.query(SuppressBug) \
+                    .filter(SuppressBug.hash == bug_to_suppress.bug_hash,
+                            SuppressBug.run_id == run_id,
+                            SuppressBug.file_name == bug_to_suppress.file_name)
```

Comment:
We are expected to have exactly one result from this query. Maybe you could use one_or_none or an assert about the length of the results to ensure the invariant.